ENIGMAsystem/SHELL/Graphics_Systems/General/GSvertex.h

```diff
@@ -67,11 +71,12 @@ int vertex_create_buffer();
 int vertex_create_buffer_ext(unsigned size);
 void vertex_delete_buffer(int buffer);
 bool vertex_exists(int buffer);
+void vertex_set_format(int buffer, int format);
 unsigned vertex_get_size(int buffer);
 unsigned vertex_get_number(int buffer);
-void vertex_freeze(int buffer);
+void vertex_freeze(int buffer, bool dynamic = false);
```

Comment:
I'm not clear on why this takes dynamic as a boolean. Also, is it worth exposing the actual background types (static, dynamic, stream) as constants, rather than just having them specify `bool dynamic`? Is it stream/dynamic by default, and then set to static/dynamic when frozen?

Reply:
I can't really say if it's worth abstracting those types. You see, `vertex_freeze` is only here for compatibility with GMS. That allegedly marks it as readonly, and I assume that means they make it a static buffer.
https://docs.yoyogames.com/source/dadiospice/002_reference/shaders/primitive%20building/vertex_freeze.html

Here's the meaning:
* `GL_STATIC_DRAW`: call `vertex_freeze` with `dynamic` false
* `GL_DYNAMIC_DRAW`: call `vertex_freeze` with `dynamic` true
* `GL_STREAM_DRAW`: don't ever call `vertex_freeze`

So that makes it compatible with GMS and has a nice parallel with how their API does it. That lets us also provide our users the ability to hint that although they are going to freeze the vertex buffer, they want to be able to update it infrequently.